executor/index_merge_reader.go

```diff
@@ -775,18 +789,217 @@ func (w *indexMergeProcessWorker) fetchLoop(ctx context.Context, fetchCh <-chan
 	}
 }
 
-func (w *indexMergeProcessWorker) handleLoopFetcherPanic(ctx context.Context, resultCh chan<- *lookupTableTask) func(r interface{}) {
+type intersectionProcessWorker struct {
+	// key: parTblIdx, val: HandleMap
+	// Value of MemAwareHandleMap is *int to avoid extra Get().
+	handleMapsPerWorker map[int]*kv.MemAwareHandleMap[*int]
+	workerID            int
+	workerCh            chan *indexMergeTableTask
+	indexMerge          *IndexMergeReaderExecutor
+	memTracker          *memory.Tracker
+	batchSize           int
+
+	// When rowDelta == memConsumeBatchSize, Consume(memUsageDelta)
+	memUsageDelta int64
+	rowDelta      int64
+
+	// For Consume(-totalMemUsage)
+	totalMemUsage int64
+}
+
+func (w *intersectionProcessWorker) consumeMemDelta() {
+	w.memTracker.Consume(w.memUsageDelta)
+	w.memUsageDelta = 0
+	w.rowDelta = 0
+}
+
+func (w *intersectionProcessWorker) doIntersectionPerPartition(ctx context.Context, workCh chan<- *indexMergeTableTask, resultCh chan<- *indexMergeTableTask, finished <-chan struct{}) {
+	defer func() {
+		if w.rowDelta > 0 {
+			w.consumeMemDelta()
+		}
+	}()
+
+	for task := range w.workerCh {
+		var ok bool
+		var hMap *kv.MemAwareHandleMap[*int]
+		if hMap, ok = w.handleMapsPerWorker[task.parTblIdx]; !ok {
+			hMap = kv.NewMemAwareHandleMap[*int]()
+			w.handleMapsPerWorker[task.parTblIdx] = hMap
+		}
+		var mapDelta int64
+		var rowDelta int64
+		for _, h := range task.handles {
+			// Use *int to avoid Get() again.
+			if cntPtr, ok := hMap.Get(h); ok {
+				(*cntPtr)++
+			} else {
+				cnt := 1
+				mapDelta += hMap.Set(h, &cnt) + int64(h.ExtraMemSize())
+				rowDelta += 1
+			}
+		}
+
+		logutil.BgLogger().Debug("intersectionProcessWorker handle tasks", zap.Int("workerID", w.workerID),
+			zap.Int("task.handles", len(task.handles)), zap.Int64("rowDelta", rowDelta))
+		if rowDelta > 0 {
+			w.rowDelta += rowDelta
+			w.memUsageDelta += mapDelta + (rowDelta * int64(unsafe.Sizeof(reflect.Int64)))
```

Comment:
Can add `(rowDelta * int64(unsafe.Sizeof(reflect.Int64)))` when `w.rowDelta >= int64(batchSize)`

Reply:
And why use reflect.Int64? Use unsafe.Sizeof(int(0))?

Reply:
D